File: StructuralPatterns/Decorator/TypeScript/NotifierDecorator.ts
import { Notifier } from './Notifier';

export class NotifierDecorator implements Notifier {
    protected wrapper: Notifier;
    constructor(wrapper: Notifier) {
        this.wrapper = wrapper;
    }
    send(message: string): void {
        this.wrapper.send(message);
    }
}

File: StructuralPatterns/Decorator/CSharp/NotifierDecorator.cs
namespace Decorator {
    public class NotifierDecorator : INotifier {
        protected INotifier wrapper;
        public NotifierDecorator(INotifier wrapper) {
            this.wrapper = wrapper;
        }
        public virtual void Send(string message) {
            wrapper.Send(message);
        }
    }
}

